src/user-management/interfaces/authentication-event.interface.ts

```diff
@@ -0,0 +1,30 @@
+interface AuthenticationEventError {
+  code: string;
+  message: string;
+}
+
+export type AuthenticationEventType =
+  | 'sso'
+  | 'password'
+  | 'oauth'
+  | 'mfa'
+  | 'magic_auth'
+  | 'email_verification';
+
+export type AuthenticationEvent = {
+  type: AuthenticationEventType;
+  email?: string | null;
+  ipAddress: string | null;
+  userAgent: string | null;
+  userId: string | null;
+  error?: AuthenticationEventError;
+};
+
+export interface AuthenticationEventResponse {
+  type: AuthenticationEventType;
+  email?: string | null;
```

Comment:
```suggestion
  email: string | null;
```

Reply:
Seems like we're not that consistent with nullable properties, there are a couple of other places that use `undefined | null`. 

I've updated `error` to `| null` as well.